### ISSUE TITLE
10.14 Munki Error 

### ISSUE THREAD

**Issue by f3xy** [munki/munki#855]
I haven't seen one yet on this issue, but a fresh Install of 10.14 and Munki generates this issue when running managedsoftwareupdate. Currently investigating possible fixes, but latest munki tools didn't seem to work (3.3.1.3525)
```
  File "/usr/local/munki/managedsoftwareupdate", line 58, in <module>
    from munkilib import appleupdates
  File "/usr/local/munki/munkilib/appleupdates/__init__.py", line 1, in <module>
    from .core import *
  File "/usr/local/munki/munkilib/appleupdates/core.py", line 22, in <module>
    from . import au
  File "/usr/local/munki/munkilib/appleupdates/au.py", line 39, in <module>
    from . import sync
  File "/usr/local/munki/munkilib/appleupdates/sync.py", line 41, in <module>
    from .. import fetch
  File "/usr/local/munki/munkilib/fetch.py", line 45, in <module>
    from . import keychain
  File "/usr/local/munki/munkilib/keychain.py", line 30, in <module>
    from OpenSSL.crypto import load_certificate, FILETYPE_PEM
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/OpenSSL/__init__.py", line 41, in <module>
    from OpenSSL import rand, SSL
ImportError: dlopen(/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/OpenSSL/SSL.so, 2): Symbol not found: _SSLv3_method
  Referenced from: /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/OpenSSL/SSL.so
  Expected in: flat namespace
 in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/OpenSSL/SSL.so
```

**Comment by clburlison** [comment]
I think this issue should be re-opened. Python on Mojave, including https requests, do work. The issue is Apple is now shipping a newer version of LibreSSL (2.6.4) in 10.14 while 10.13 was linked against (2.2.7). During this upgrade, the maintainers of LibereSSL have removed support for SSL v3 which causes the above import error.

As far as I can see we only import from OpenSSL once in the `keychain.py` file and use the `load_certificate` function

https://github.com/munki/munki/blob/e05b5798b675581d386e6f7ee587ba69f637d4da/code/client/munkilib/keychain.py#L157-L159

Since we are simply using this call to get the common name we should be able to swap this out with a subprocess call to `/usr/bin/openssl` to return the common name. A hacked together bash/sed version 

```bash
openssl x509 -noout -subject -in ~/Desktop/my_cert.pem | sed -n '/^subject/s/^.*CN=//p'
```

based off a few local tests this should be backwards compatible.

**Comment by f3xy** [comment]
I can re-open it. I just noticed this doing some initial testing and I'm well aware not to expect a fix anytime soon. I just didn't want this to come off as "pls fix".

**Comment by gregneagle** [comment]
Addressed by #856
